src/pfe/file-watcher/server/src/projects/OdoExtensionProject.ts

```diff
@@ -0,0 +1,275 @@
+        projectUtil.containerUpdate(operation, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "update");
+    }
+
+    /**
+     * @function
+     * @description Rebuild operation for the project.
+     *
+     * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
+     */
+    rebuild = async (projectInfo: ProjectInfo): Promise<void> => {
+        await projectUtil.runScript(projectInfo, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "rebuild");
+    }
+
+    /**
+     * @function
+     * @description Delete operation for the project.
+     *
+     * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
+     */
+    deleteContainer = async (projectInfo: ProjectInfo): Promise<void> => {
+        await projectUtil.containerDelete(projectInfo, path.join(this.fullPath, "odo-extension-entrypoint.sh"));
+    }
+
+    /**
+     * @function
+     * @description Validate project.
+     *
+     * @param operation <Required | Operation> - The validate operation for the project.
+     */
+    validate = async (operation: Operation): Promise<void> => {
+        const validator = new Validator(operation);
+        await validator.validateRequiredFiles(this.config.requiredFiles);
+        validator.sendResult();
+    }
+
+    /**
+     * @function
+     * @description Get the build log path for the project.
+     *
+     * @param logDirectory <Required | string> - The log location directory.
+     *
+     * @returns Promise<BuildLog>
+     */
+    getBuildLog = async (logDirectory: string): Promise<BuildLog> => {
+        return await logHelper.getBuildLogs(logDirectory, this.config.buildLogs);
+    }
+
+    /**
+     * @function
+     * @description Get the app log path for the project.
+     *
+     * @param logDirectory <Required | String> - The log location directory.
+     *
+     * @returns Promise<AppLog>
+     */
+    getAppLog = async (logDirectory: string): Promise<AppLog> => {
+        return await logHelper.getAppLogs(logDirectory, this.config.appLogs);
+    }
+
+    /**
+     * @function
+     * @description Get the project capabilities of the project.
+     *
+     * @returns ProjectCapabilities
+     */
+    getCapabilities = (): ProjectCapabilities => {
+
+        if (this.config.capabilities) {
+            return new ProjectCapabilities(this.config.capabilities.startModes, this.config.capabilities.controlCommands);
+        }
+
+        return defaultProjectCapabilities;
+    }
+
+    /**
+     * @function
+     * @description Return a reproducible container name for the project.
+     *
+     * @param projectID <Required | string> - An alphanumeric identifier for a project.
+     * @param projectLocation <Required | string> - The project location directory.
+     *
+     * @returns Promise<string>
+     */
+    getContainerName = async (projectID: string, projectLocation: string): Promise<string> => {
```

Comment:
If I remember correctly, we discussed before that we want to have the container name starts with "cw-". Please add.

Reply:
Yes, it's already added, actually the ODO "getContainerName" is different from other project types "getContainerName", other project types can specify container name through "getContainerName", but ODO can't due to ODO will generate container name automatically, we can only specify the component name, so I add the `cw-` here: https://github.com/eclipse/codewind/pull/601/files#diff-7597df0b79304bd2be18940009414316R800